common/djangoapps/entitlements/api/v1/tests/test_views.py

```diff
@@ -141,6 +141,32 @@ def test_add_entitlement(self):
         )
         assert results == CourseEntitlementSerializer(course_entitlement).data
 
+    def test_add_duplicate_entitlement(self):
+        """
+        Request with identical course_uuid and order_number should not create duplicate
+        entitlement
+        """
+        course_uuid = uuid.uuid4()
+        entitlement_data = self._get_data_set(self.user, str(course_uuid))
+
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 201
+        response = self.client.post(
+            self.entitlements_list_url,
+            data=json.dumps(entitlement_data),
+            content_type='application/json',
+        )
+        assert response.status_code == 400
+        course_entitlement = CourseEntitlement.objects.filter(
+            course_uuid=course_uuid,
+            order_number=entitlement_data['order_number']
+        )
+        self.assertEqual(len(course_entitlement), 1)
```

Comment:
I would consider using one convention for equality testing.  i.e. `assert len(course_entitlement) == 1`